Assets/_BForBoss/_Core/Scripts/Managers/WaveManager.cs

```diff
@@ -0,0 +1,93 @@
+using System;
+using System.Collections;
+using Sirenix.OdinInspector;
+using UnityEngine;
+
+namespace BForBoss
+{
+    public class WaveManager : MonoBehaviour
+    {
+        [Title("Components")]
+        [SerializeField] private EnemySpawnerManager _enemySpawnerManager;
+        
+        [Title("Settings")] 
+        [SerializeField, Tooltip("Number of enemies for the first wave")] private int _initialNumberOfEnemies = 10;
+        [SerializeField, Tooltip("Cooldown time between waves")] private float _timeBetweenWaves = 2.5f;
+        [SerializeField, Tooltip("Number of enemies per wave is the number of enemies from the previous wave multiplied by this multiplier")] private float _EnemyAmountMultiplier = 1.2f;
+
+        private int _maxCountForCurrentWave = 0;
+        private int _spawnCount = 0;
+        private int _killCount = 0;
+        private int _waveNumber = 1;
+
+        public void Initialize(LifeCycleManager lifeCycleManager)
+        {
+            _maxCountForCurrentWave = _initialNumberOfEnemies;
+            _enemySpawnerManager.Initialize(lifeCycleManager, OnEnemySpawned, OnEnemyKilled);
+        }
+
+        public void Reset()
+        {
+            _maxCountForCurrentWave = _initialNumberOfEnemies;
+            _spawnCount = 0;
+            _killCount = 0;
+            _waveNumber = 1;
+
+            _enemySpawnerManager.Reset();
+        }
+
+        private void OnEnemySpawned()
+        {
+            _spawnCount++;
+
+            if (_spawnCount >= _maxCountForCurrentWave)
+            {
+                PauseSpawning();
+            }
+        }
+
+        private void OnEnemyKilled()
+        {
+            _killCount++;
+            
+            //Added the second clause just in case no. spawned > max allowed
+            if (_killCount >= _maxCountForCurrentWave && _killCount == _spawnCount)
+            {
+                _enemySpawnerManager.Reset();
+                Debug.Log($"<b>Wave {_waveNumber}</b> Ended. Please wait <b>{_timeBetweenWaves} seconds</b> before the next wave");
```

Comment:
Leaving Debug comments here, for now, to indicate when a wave has been completed, and when it has started (for testing purposes). Can remove when UI comes into place